### ISSUE TITLE
[FEAT] Implement model-agnostic BayesianTuner with LGBMTuner preset

### ISSUE THREAD

**Issue by Cristiano2132** [Cristiano2132/model-track-cr#58]
## 📋 Context

The current `BayesianTuner` logic is scattered across notebooks (hardcoded for LightGBM binary classification). The `tuning/` module centralizes this into a **model-agnostic, task-aware** tuner.

**Design goals:**
1. Works with any model that follows the sklearn `fit(X, y)` / `predict_proba(X)` interface
2. Task-aware: metric for optimization selected based on `TaskType` (AUC for binary, macro-AUC for multiclass, RMSE for regression)
3. LightGBM-first: ships with `LGBMTuner` preset, extensible to XGBoost, CatBoost, etc.

## 🎯 Class Design

```mermaid
classDiagram
    class BaseTuner {
        <<abstract>>
        +task: TaskAdapter
        +param_bounds: dict
        +cv_folds: int = 3
        +n_iter: int = 15
        +init_points: int = 5
        +best_params_: dict | None
        +best_score_: float | None
        +tune(X_train, y_train) dict
        +get_model() Any
    }

    class BayesianTuner {
        +task: TaskAdapter = BinaryAdapter()
        +tune(X_train, y_train) dict
        +get_model() estimator
    }

    class LGBMTuner {
        +DEFAULT_BOUNDS: dict
        +tune(X_train, y_train) dict
        +get_model() LGBMClassifier | LGBMRegressor
    }

    BaseTuner <|-- BayesianTuner
    BayesianTuner <|-- LGBMTuner
```

### Default param bounds per model type

```python
LGBMTuner.DEFAULT_BOUNDS = {
    "max_depth": (3, 20),
    "num_leaves": (20, 300),
    "learning_rate": (0.01, 0.3),
    "feature_fraction": (0.3, 0.9),
    "bagging_fraction": (0.3, 0.9),
    "lambda_l1": (0, 5),
    "lambda_l2": (0, 5),
}
```

### Usage

```python
from model_track.tuning import LGBMTuner
from model_track.base import BinaryAdapter

tuner = LGBMTuner(task=BinaryAdapter(), cv_folds=3, n_iter=20)
best_params = tuner.tune(X_train, y_train)

model = tuner.get_model()
model.fit(X_train, y_train)
```

### Task-aware metric selection

| `TaskType` | Optimization metric |
|------------|---------------------|
| `BINARY` | ROC AUC |
| `MULTICLASS` | Macro AUC (OvR) |
| `REGRESSION` | Negative RMSE |

### Extensibility (future models)

```python
class XGBTuner(BayesianTuner):
    DEFAULT_BOUNDS = {...}
    def get_model(self): return XGBClassifier(**self.best_params_)
```

## ✅ Acceptance Criteria

- [x] `BaseTuner` ABC defined
- [x] `BayesianTuner` wraps `bayesian-optimization` library
- [x] `LGBMTuner` preset with sensible defaults for binary, multiclass, regression
- [x] Task-aware metric selection via `TaskAdapter.default_metrics()[0]`
- [x] `tuning` extras group in `pyproject.toml` (lightgbm + bayesian-optimization)
- [x] Works without `tuning` extras installed (graceful `ImportError` with clear message)
- [x] Exported from `model_track.tuning`
- [x] Unit tests (mocked optimizer to avoid long runs)
- [x] Integration test: full tune → fit → evaluate flow with LightGBM

## 📁 New Files

- `src/model_track/tuning/__init__.py`
- `src/model_track/tuning/base.py`
- `src/model_track/tuning/bayesian.py`
- `src/model_track/tuning/lgbm.py`
- `tests/unit/tuning/test_lgbm_tuner.py`
- `tests/integration/test_tuning_flow.py`

**Comment by Cristiano2132** [comment]
## ✅ Implementação Concluída

- **PR mergeado:** #85
- **Branch base:** develop
- **Critérios de aceitação:** todos marcados ✅
- **Cobertura de testes:** verificada (98.13% total, >90% no módulo tuning)
- **Linting/Mypy:** passou sem erros
- **SonarCloud:** Quality Gate atingido com sucesso
